### ISSUE TITLE
selenium - test 1+1 in sagews

### ISSUE THREAD

**Issue by DrXyzzy** [sagemathinc/smc#1366]
Look at @haraldschilly  [selenium branch](https://github.com/sagemathinc/smc/tree/selenium) and get it so far to automatically compute 1+1 in a sagews. It's 90% there, what's missing is to control codemirror in a sagews cell.

**Comment by williamstein** [comment]
for what it is worth... I've been writing tons of testing code (in mocha) for the database work this last week... and boy do I **love testing** existing.  Testing asynchronous code is an interesting challenge...

**Comment by DrXyzzy** [comment]
And I am hooked on pytest for sagews. Far less chance of my introducing silly bugs.

**Comment by williamstein** [comment]
> hooked ...

Yep, addicting is a pretty good way to describe automated testing.

**Comment by haraldschilly** [comment]
the crazy thing is, the selenium setup is close to being useful for pytest.
the fun part is to extract a class+API from that proof of work script I've
done, such that the whole front end can be controlled on that level. I am
on my phone now, but if Ib don't forget I'll write about this plan here...
